src/rules/verification.utils.ts

```diff
@@ -0,0 +1,107 @@
+import { Response } from '@/types'
+import { VerificationRule } from '@/types/rules'
+import { exhaustive } from '@/utils/typescript'
+
+const NEWLINE_REGEX = /(?:\r\n|\r|\n)/g
+
+export function getValueFromRule(rule: VerificationRule, response: Response) {
+  const { type } = rule.value
+
+  switch (type) {
+    case 'recordedValue':
+      return getRecordedValue(rule.target, response)
+    case 'string':
+      return `'${rule.value.value}'`
+    case 'variable':
+      return `VARS['${rule.value.variableName}']`
+    default:
+      return exhaustive(type)
+  }
+}
+
+export function getCheckExpression(
+  rule: VerificationRule,
+  value: string | number
+): string {
+  const target = getTarget(rule)
+
+  switch (rule.operator) {
+    case 'equals':
+      return `${target} === ${value}`
+    case 'contains':
+      return `${target}.includes(${value})`
+    case 'notContains':
+      return `!${target}.includes(${value})`
+    default:
+      return exhaustive(rule.operator)
+  }
+}
+
+export function getCheckDescription(rule: VerificationRule): string {
+  const valueDescription = getValueDescription(rule)
+
+  switch (rule.operator) {
+    case 'equals':
+      return `${rule.target} equals ${valueDescription}`
+    case 'contains':
+      return `${rule.target} contains ${valueDescription}`
+    case 'notContains':
+      return `${rule.target} does not contain ${valueDescription}`
+    default:
+      return exhaustive(rule.operator)
+  }
+}
+
+function getRecordedValue(
+  target: VerificationRule['target'],
+  response: Response
+) {
+  switch (target) {
+    case 'status':
+      return `'${response.statusCode}'`
+    case 'body': {
+      // Remove newlines when comparing the body to a recorded value
+      const singleLineContent = response.content.replace(NEWLINE_REGEX, '')
+      const escapedContent = escapeBackticksAndDollarSign(singleLineContent)
+
+      return `String.raw\`${escapedContent}\``
```

Comment:
Ideally, we would save responses in separate file, import and reference them in checks. This would add some additional challenges managing the imports, as the first iteration I'm adding the whole response body as a single-line string. 

This might be easier to solve when/if we switch to AST 🌲